envoy/router/path_match_policy.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Used to decide if path match is needed based on the target route.
+ * Subclassing Logger::Loggable so that implementations can log details.
+ */
+class PathMatchPredicate : Logger::Loggable<Logger::Id::router> {
+public:
+  PathMatchPredicate() = default;
+  virtual ~PathMatchPredicate() = default;
+
+  /**
+   * @return the name of the current predicate.
+   */
+  virtual std::string name() const PURE;
+
+  /**
+   * Used to determine if the current url matches the predicate pattern.
+   *
+   * @param pattern current url of route
```

Comment:
is it the literal url or a regex pattern?  let's make sure naming makes sense

Reply:
changed